interface/language/lang_definition.php

```diff
@@ -57,9 +57,10 @@
 <?php
 
 // set up the mysql collation string to ensure case is sensitive (or insensitive) in the mysql queries
+
 if (!$disable_utf8_flag) {
   $case_sensitive_collation = "COLLATE utf8_bin";
-  $case_insensitive_collation = "COLLATE utf8_general_ci";
+  $case_insensitive_collation = $mysqli->character_set_name();
```

Comment:
You are right.  Should that not be something that the controller grabs for session data, or am I still thinking too small and too late?  Would that not be better than checking it here every time?
I imaging that the libreehr user has permissions to ask the database what the collation is.